src/frontend/src/views/Upload/components/Review/index.tsx

```diff
@@ -31,19 +31,53 @@ import {
   StyledButton,
   StyledCheckbox,
 } from "./style";
+import {
+  AnalyticsUploadMetadataType,
+  EVENT_TYPES,
+  UploadFormMetadataType,
+} from "src/common/analytics/eventTypes";
+import { analyticsTrackEvent } from "src/common/analytics/methods";
 
 export default function Review({
   samples,
   metadata,
   cancelPrompt,
   analyticsFlowUuid,
+  hasManuallyEditedMetadata,
+  hasImportedMetadataFile,
 }: Props): JSX.Element {
   const pathogen = useSelector(selectCurrentPathogen);
   const { data: userInfo } = useUserInfo();
   const [isGroupConfirmationChecked, setIsGroupConfirmationChecked] =
     useState<boolean>(false);
   const [isConsentChecked, setIsConsentChecked] = useState(false);
 
+  let numberOfDetectedSamples = 0;
+  if (samples != null) {
+    numberOfDetectedSamples = Object.keys(samples).length;
+  }
+
+  // firing off analytics event in review since metadata editing should be finalized at this point.
+  useEffect(() => {
+    const hasMetadataBeenEdited =
```

Comment:
If I'm understanding it correctly, this conditional should not be needed. I don't think a person can get to the review step without either uploading a file or manually entering metdata.

Reply:
I wish, but you can actually refresh the page (or directly navigate via URL) while you're in the middle of the Upload flow and you get to that page of the Upload flow but with no data. It's arguable whether the defensive coding approach is necessary to cover this corner case, but I think it doesn't hurt to have it.

Reply:
yeah :/